### PR TITLE
TERA-192: Add authenticated user to the http request to allow the use…

### PR DESCRIPTION
**Pull request by thomas-anderson-bsl**
…rname to be part of the request log

### CODE REVIEW

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AddUserAuthToRequestFilter.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @param <P> the type of principal retrieved from authentication
+ * @author brightSPARK Labs
+ */
+public class AddUserAuthToRequestFilter<P extends Principal> implements ContainerRequestFilter {
+
+    private final PrincipalConverter<P> principalConverter;
+
+    public AddUserAuthToRequestFilter(final PrincipalConverter<P> principalConverter) {
+        this.principalConverter = principalConverter;
```
Comment (michael-forman) [inline src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AddUserAuthToRequestFilter.java]: requireNonNull?

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/PrincipalConverter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.security.Principal;
+
+/**
+ * Interface for converting between {@link InternalUser} and the {@link Principal} used in the
+ * system.
+ *
+ * @param <P> The {@link Principal} type to convert
+ */
+public interface PrincipalConverter<P extends Principal> {
+    public InternalUser convertToInternalUser(P principal);
```
Comment (michael-forman) [inline src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/PrincipalConverter.java]: public is redundant for interface methods

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AddUserAuthToRequestFilter.java (hunk 1 above)
Comment (thomas-anderson-bsl) [reply src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/AddUserAuthToRequestFilter.java]: :+1:

File: src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/PrincipalConverter.java (hunk 2 above)
Comment (thomas-anderson-bsl) [reply src/main/java/com/brightsparklabs/dropwizard/bundles/auth/external/PrincipalConverter.java]: :+1:  also added javadoc

**Pull request merged**